tests/signing/test_signing.py

```diff
@@ -0,0 +1,329 @@
+
+        # verify the file was decrypted
+        self.assertTrue(result)
+        self.assertTrue(os.path.isfile(decrypted_file))
+
+        # verify the contents
+        with codecs.open(decrypted_file, 'r', encoding='utf-8') as in_file:
+            decrypted_text = in_file.read()
+
+        with codecs.open(os.path.join(self.resources_dir, 'test.txt'), 'r', encoding='utf-8') as in_file:
+            original_text = in_file.read()
+
+        self.assertEqual(decrypted_text, original_text)
+
+    def test_sign_file_with_test_certificate(self):
+
+        # initialization
+        source_file = os.path.join(self.temp_dir, 'source.json')
+        sig_file = os.path.join(self.temp_dir, 'source.sig')
+
+        # copy test file to the temp directory
+        shutil.copy(os.path.join(self.resources_dir, 'source.json'), source_file)
+
+        # check that the source file exists in the temp directory
+        self.assertTrue(os.path.isfile(source_file))
+
+        # check that .sig file DOES NOT exist
+        self.assertFalse(os.path.isfile(sig_file))
+
+        # sign the file
+        sig_file_name = Signing.sign_file(source_file,
+                                          pem_file=os.path.join(self.resources_dir, 'unit-test-private.pem'))
+
+        # check that .sig file DOES exist now
+        self.assertEqual(sig_file, sig_file_name)
+        self.assertTrue(os.path.isfile(sig_file))
+
+        # verify the .sig file is correct
+        self.assertTrue(Signing.verify_signature(source_file, sig_file_name,
+                                                 pem_file=os.path.join(self.resources_dir, 'unit-test-public.pem')))
+
+    @unittest.skipIf(Signing.is_travis(), 'Skipping test_sign_file_with_live_certificate on Travis CI.')
+    def test_sign_file_with_live_certificate(self):
+
+        # initialization
+        source_file = os.path.join(self.temp_dir, 'source.json')
+        sig_file = os.path.join(self.temp_dir, 'source.sig')
+
+        # copy test file to the temp directory
+        shutil.copy(os.path.join(self.resources_dir, 'source.json'), source_file)
+
+        # check that the source file exists in the temp directory
+        self.assertTrue(os.path.isfile(source_file))
+
+        # check that .sig file DOES NOT exist
+        self.assertFalse(os.path.isfile(sig_file))
+
+        # sign the file
+        sig_file_name = Signing.sign_file(source_file)
+
+        # check that .sig file DOES exist now
+        self.assertEqual(sig_file, sig_file_name)
+        self.assertTrue(os.path.isfile(sig_file))
+
+        # verify the .sig file is correct
+        self.assertTrue(Signing.verify_signature(source_file, sig_file_name))
+
+    def test_get_default_pem_file(self):
+
+        if Signing.is_travis():
+            with self.assertRaises(Exception) as context:
+                Signing.get_default_pem_file()
+
+                self.assertEqual('Not able to decrypt the pem file.', str(context.exception))
```

Comment:
This line needs to be moved outside of the `with` block  ([docs here](https://docs.python.org/2.7/library/unittest.html#unittest.TestCase.assertRaises)).